### ISSUE TITLE
Protobuf-c does not produce usable output

### ISSUE THREAD

**Issue by baoshi** [Homebrew/homebrew-core#168501]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
HOMEBREW_VERSION: 4.2.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0476c2e5e4f02b409084553896340b2f4ab3352e
Last commit: 3 days ago
Core tap JSON: 10 Apr 03:17 UTC
Core cask tap JSON: 10 Apr 03:17 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.44.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I'm using protobuf-c in my development. A recent update rendered protobuf-c unusable.

### What happened (include all command output)?

The recent proto-c command gives the error below:
```
protoc-c --c_out=. test.proto
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1712719354.625982 8260452 command_line_interface.cc:1534] Built-in generator --c_out specifies a maximum edition PROTO3 which is not the protoc maximum 2023.
```

### What did you expect to happen?

It should generate device.pb-c.h and device.pb-c.c as old version does, but the error above occurs.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1. Install protobuf-c using brew install protobuf-c. Current version shows `1.5.0_4`
2. Create a test.proto file as following (same as the protobuf-c.rb file is using) 

syntax = "proto3";
package test;
message TestCase {
  string name = 4;
}
message Test {
  repeated TestCase case = 1;
}
```
3. Run protoc-c --c_out=. test.proto
4. Error shows like above.
```

**Comment by SMillerDev** [comment]
If those are optional fields, that is https://github.com/protobuf-c/protobuf-c/issues/476.

Either way, since we can see from the reproduction instructions that this isn't brew specific I will close this. You're better off filling the bug in https://github.com/protobuf-c/protobuf-c/issues if it's not the one above.

**Comment by ZhongRuoyu** [comment]
This seems to be a compatibility issue with the latest Protobuf 26.1, as discussed in https://github.com/protobuf-c/protobuf-c/pull/711#issuecomment-2045527121.

I believe you can invoke `protoc --c_out=...` instead of `protoc-c --c_out=...`, as a workaround.

**Comment by carlocab** [comment]
Will be fixed in 06792033f7e9660d18796bdcfe015e5da71443fa. When that's merged, do `brew update && brew reinstall protobuf-c`.
